lib/web/conn_upgrade.go

```diff
@@ -88,6 +93,41 @@ func (h *Handler) upgradeALPN(ctx context.Context, conn net.Conn) error {
 	return h.cfg.ALPNHandler(ctx, waitConn)
 }
 
+func (h *Handler) upgradeALPNWithPing(ctx context.Context, conn net.Conn) error {
+	if h.cfg.ALPNHandler == nil {
+		return trace.BadParameter("missing ALPNHandler")
+	}
+
+	pingConn := pingconn.New(conn)
+
+	// Cancel ping background goroutine when connection is closed.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	go h.startPing(ctx, pingConn)
+	time.Sleep(time.Millisecond * 100)
```

Comment:
We should use ProxyConfig.PingInterval https://github.com/gravitational/teleport/blob/922fc8bfb4046f9617bed2ed007e57a53a7c40ca/lib/srv/alpnproxy/proxy.go#L68